src/Kernel/client/ExecutionPathVisualizer/pathVisualizer.ts

```diff
@@ -0,0 +1,121 @@
+import { formatInputs } from "./formatters/inputFormatter";
+import { formatGates } from "./formatters/gateFormatter";
+import { formatRegisters } from "./formatters/registerFormatter";
+import { processOperations } from "./process";
+import { ExecutionPath } from "./executionPath";
+import { Metadata } from "./metadata";
+import { GateType } from "./constants";
+
+const script = `
+<script type="text/JavaScript">
+    function toggleClassicalBtn(cls) {
+        const textSvg = document.querySelector(\`.\${cls} text\`);
+        const group = document.querySelector(\`.\${cls}-group\`);
+        const currValue = textSvg.childNodes[0].nodeValue;
+        const zeroGates = document.querySelector(\`.\${cls}-zero\`);
+        const oneGates = document.querySelector(\`.\${cls}-one\`);
+        switch (currValue) {
+            case '?':
+                textSvg.childNodes[0].nodeValue = '1';
+                group.classList.remove('cls-control-unknown');
+                group.classList.add('cls-control-one');
+                break;
+            case '1':
+                textSvg.childNodes[0].nodeValue = '0';
+                group.classList.remove('cls-control-one');
+                group.classList.add('cls-control-zero');
+                oneGates.classList.toggle('hidden');
+                zeroGates.classList.toggle('hidden');
+                break;
+            case '0':
+                textSvg.childNodes[0].nodeValue = '?';
+                group.classList.remove('cls-control-zero');
+                group.classList.add('cls-control-unknown');
+                zeroGates.classList.toggle('hidden');
+                oneGates.classList.toggle('hidden');
+                break;
+        }
+    }
+</script>
+`;
+
+const style = `
+<style>
+    .hidden {
+        display: none;
+    }
+    .cls-control-unknown {
+        opacity: 0.25;
+    }
+    <!-- Gate outline -->
+    .cls-control-one rect,
+    .cls-control-one line,
+    .cls-control-one circle {
+        stroke: #4059bd;
+        stroke-width: 1.3;
+    }
+    .cls-control-zero rect,
+    .cls-control-zero line,
+    .cls-control-zero circle {
+        stroke: #c40000;
+        stroke-width: 1.3;
+    }
+    <!-- Gate label -->
+    .cls-control-one text {
+        fill: #4059bd;
+    }
+    .cls-control-zero text {
+        fill: #c40000;
+    }
+    <!-- Control button -->
+    .cls-control-btn {
+        cursor: pointer;
+    }
+    .cls-control-unknown .cls-control-btn {
+        fill: #e5e5e5;
+    }
+    .cls-control-one .cls-control-btn {
+        fill: #4059bd;
+    }
+    .cls-control-zero .cls-control-btn {
+        fill: #c40000;
+    }
+    <!-- Control button text -->
+    .cls-control-unknown .cls-control-text {
+        fill: black;
+        stroke: none;
+    }
+    .cls-control-one .cls-control-text,
+    .cls-control-zero .cls-control-text {
+        fill: white;
+        stroke: none;
+    }
```

Comment:
Do you know how these colors behave under different system settings, e.g. high contrast mode?

Reply:
Not really, what are the normal procedures for adding colours in?

Reply:
I'm not aware of any place in IQ# where we are specifying colors in this way; you may be breaking new ground. 😊 You could try it out quickly by just turning on High Contrast mode (in Windows 10, if you click Start and type "High Contrast" it should take you there). Basically we just want to make sure that accessibility settings like High Contrast are respected by our features. I am guessing that the browser will do approximately the right thing here, but it's worth testing.

Reply:
I'd also suggest using Edge / Chrome extensions to simulate common forms of colorblindness can help with accessibility as well.

Reply:
Thanks for the feedback! I installed a chrome extension called "Colorblinding" (couldn't find an edge one) and the only case where the colours between "true" and "false" were not differentiable was with the "monochromatic" setting.